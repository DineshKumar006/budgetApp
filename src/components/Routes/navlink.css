.is-active{
    color: green;
    font-weight: bold;
}
.active{
    color:black;
}
