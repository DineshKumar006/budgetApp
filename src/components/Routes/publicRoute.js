import {Route,Redirect} from 'react-router-dom';
import {connect} from 'react-redux'
import React  from 'react'

 const PublicRoute=({isAuthenticated,component:Component, ...rest})=>{
    //  console.log(...rest)
    return(
        <div>
            <Route  {...rest} component={(props)=>(

         isAuthenticated?
              
                    <Redirect to="/dashboard"/>
                   
             :  <Component {...props}/>

            )}/>

        </div>
    )
}


const mapStateToProps=(state)=>{
    return{
        isAuthenticated : !! state.AuthReducer.uid
    }
}

export default connect(mapStateToProps)(PublicRoute)

