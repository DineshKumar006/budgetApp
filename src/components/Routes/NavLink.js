import React from 'react'
import {NavLink} from 'react-router-dom'
import './navlink.css'
const Navlink=(props)=>{
    return(
        <div>
                <NavLink to={props.link} activeClassName={'is-active'} exact={true}> {props.children}</NavLink>
        </div>
    )
}

export default Navlink