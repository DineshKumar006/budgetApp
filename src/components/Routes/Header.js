import React from 'react';

import NavLink from './NavLink'
 import './navlink.css'
import {LogoutStart} from '../../ReduxStore/actions/Auth'
import {connect} from 'react-redux'

import {createBrowserHistory} from 'history';


export const history=createBrowserHistory()

const Header=(props)=>{
    return(
        <React.Fragment>
        <div>
            <h1 className={'header'}>Budget App.</h1>
        </div>
        <header>
            
        <button onClick={props.LogoutAuth}>Logout</button>
            <div >

            <NavLink link='/dashboard' ><strong>Dashboard</strong></NavLink>
            <NavLink  link="/CreateExpenses"><strong>CreateExpenses</strong></NavLink>
            {/* <NavLink link="/edit/:id"></NavLink> */}
            </div>
           

        </header>
        </React.Fragment>
        
    )
}

const mapDispatchToProps=(dispatch)=>{
    return{
        LogoutAuth:()=>dispatch(LogoutStart())
    }
}



export default connect(undefined,mapDispatchToProps)(Header)