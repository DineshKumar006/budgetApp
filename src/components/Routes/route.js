import React from 'react';
import {Route,Switch} from 'react-router-dom';
import  Home from '../home/Home';
import Login from '../login/login'
import NavLink from './NavLink'
import PageNotFound from '../404Page/404Page'
 import './navlink.css'
import AddExpenses from '../expenses/AddExpenses'
import EditExpenses from '../expenses/EditExpense'

export const Header=()=>{
    return(
        <header>
            <h1>Budget-App</h1>
            
            <div >
            <NavLink link='/' ><strong>Dashboard</strong></NavLink>
            <NavLink link='/login'><strong>Login</strong></NavLink>
            <NavLink  link="/CreateExpenses"><strong>CreateExpenses</strong></NavLink>
            <NavLink link="/edit/:id"></NavLink>
            </div>
           

        </header>
    )
}


const Routes=(props)=>{
    return(
        <div>

    <div>
        <Header/>
       
        <Switch>
            <Route path='/' exact={true}  component={Home}/>
            <Route path='/CreateExpenses' exact={true} component={AddExpenses}/>
            <Route path='/login' exact={true}  component={Login}/>
            {/* <Route path='/login/:id' exact={true}  component={Login}/> */}
            <Route path="/edit/:id" exact={true} component={EditExpenses}/>
            <Route  component={PageNotFound}  />
        </Switch>
    </div>
       
        </div>
    )
}

export default Routes