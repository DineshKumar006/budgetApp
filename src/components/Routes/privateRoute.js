import {Route,Redirect} from 'react-router-dom';
import {connect} from 'react-redux'
import React  from 'react'
import Header from './Header'

 const PrivateRoute=({isAuthenticated,component:Component, ...rest})=>{
     console.log(...rest)
    return(
        <div>
            <Route  {...rest} component={(props)=>(

         isAuthenticated?
                (<div>
                    <Header/>
                    <Component {...props}/>
                </div>) : <Redirect to="/"/>

            )}/>

        </div>
    )
}


const mapStateToProps=(state)=>{
    return{
        isAuthenticated : !! state.AuthReducer.uid
    }
}

export default connect(mapStateToProps)(PrivateRoute)

