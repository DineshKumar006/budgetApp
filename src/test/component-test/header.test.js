import React from 'react'


import {Header} from '../../components/Routes/route';
// import ReactShallowRenderer from 'react-test-renderer/shallow'
import {shallow} from 'enzyme'
import toJSON, { toJson } from 'enzyme-to-json'


test('Run header Compomnent',()=>{

    const wrapper=shallow(<Header/>)
    // expect(wrapper.find('h1').text()).toBe('Budget-App')

    expect(toJSON(wrapper)).toMatchSnapshot()

    // const renderer=new ReactShallowRenderer();
    // renderer.render(<Header/>)
    // expect(renderer.getRenderOutput()).toMatchSnapshot();
    // console.log(renderer.getRenderOutput())
})
