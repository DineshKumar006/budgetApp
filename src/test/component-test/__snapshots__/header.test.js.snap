// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Run header Compomnent 1`] = `
<header>
  <h1>
    Budget-App
  </h1>
  <div>
    <Navlink
      link="/"
    >
      <strong>
        Dashboard
      </strong>
    </Navlink>
    <Navlink
      link="/login"
    >
      <strong>
        Login
      </strong>
    </Navlink>
    <Navlink
      link="/CreateExpenses"
    >
      <strong>
        CreateExpenses
      </strong>
    </Navlink>
    <Navlink
      link="/edit/:id"
    />
  </div>
</header>
`;
